File: PLT/SampleCode/FindShoot.r
think

   closest_enemy! = NOTHING!
   // 1. Get enemy locations
   ping_radar  // Decreases energy
   enemies!... = get_enemies  // alternatively: enemies!... = enemies!... of radar

   // 2. Find closest
   closest_enemy! = find_closest_enemy enemies!...

   // 3. Shoot
   shoot closest_enemy!  // Decreases energy

end

instruction find_closest_enemy with enemies!... means

   closest_distance# = MAX#
   closest_enemy! = NOTHING!
   my_loc@ = location@ of self

   repeat with each enemy! in enemies!...

      enemy_loc@ = location of enemy!
      distance# = distance enemy_loc@ my_loc@
   
      if distance# is_less_than closest_distance# then
      
         closest_distance# = distance#
         closest_enemy! = enemy!
         
      done

   done

gives closest_enemy!

File: trunk/PLT/SampleCode/FindShoot.java
package app;

import java.util.*;

public class RobotCompiled extends Robot {

	public static void think() {

		Enemy closest_enemy = new Enemy();
		ping_radar();
		LinkList<Enemy> enemies = get_enemies();
		closest_enemy = find_closest_enemy(enemies);
		shoot(closest_enemy);

	}


	public static Enemy find_closest_enemy(LinkList<Enemy> enemies) {

		float closest_distance = Float.MAX_VALUE;
		Enemy closest_enemy = new Enemy();
		Location my_loc = this.location;
		
		for (Enemy enemy : enemies) {

			Location enemy_loc = enemy.location;
			float distance = distance(enemy_loc, my_loc);
			if(distance<=closest_distance){
			closest_distance = distance;
			closest_enemy = enemy;

			}


		}



	}
}
